common/config.go

```diff
@@ -22,28 +22,49 @@ const (
 var homeDir, _ = os.UserHomeDir()
 var baseDir = path.Join(homeDir, ".atomicswap")
 
+// MoneroNode represents the host and port of monerod's RPC endpoint
+type MoneroNode struct {
+	Host string
+	Port uint
+}
+
 // Config contains constants that are defaults for various environments
 type Config struct {
-	DataDir          string
-	MoneroDaemonHost string
-	MoneroDaemonPort uint
-	ContractAddress  ethcommon.Address
-	Bootnodes        []string
+	DataDir         string
+	MoneroNodes     []*MoneroNode
+	ContractAddress ethcommon.Address
+	Bootnodes       []string
 }
 
 // MainnetConfig is the mainnet ethereum and monero configuration
 var MainnetConfig = Config{
-	DataDir:          path.Join(baseDir, "mainnet"),
-	MoneroDaemonHost: "127.0.0.1",
-	MoneroDaemonPort: DefaultMoneroDaemonMainnetPort,
+	DataDir: path.Join(baseDir, "mainnet"),
+	MoneroNodes: []*MoneroNode{
+		{
+			Host: "127.0.0.1",
+			Port: DefaultMoneroDaemonMainnetPort,
+		},
+	},
 }
 
 // StagenetConfig is the monero stagenet and ethereum Gorli configuration
 var StagenetConfig = Config{
-	DataDir:          path.Join(baseDir, "stagenet"),
-	MoneroDaemonHost: "node.sethforprivacy.com",
-	MoneroDaemonPort: 38089, // Seth is not using the default stagenet value of 38081 (so don't use our constant)
-	ContractAddress:  ethcommon.HexToAddress("0xd2B5d6252D0645E4cF4Bb547E82A485F527BEFb7"),
+	DataDir: path.Join(baseDir, "stagenet"),
+	MoneroNodes: []*MoneroNode{
+		{
+			Host: "node.sethforprivacy.com",
+			Port: 38089,
+		},
+		{
+			Host: "node.monerodevs.org",
+			Port: 38089,
+		},
+		{
+			Host: "stagenet.community.rino.io",
+			Port: 38081,
+		},
```

Comment:
I just pulled some nods from here: https://monero.fail/?nettype=stagenet
Suggestions on the best ones to pick are welcome.